### PR TITLE
fix deepcopy for allOf schemas

### PR DESCRIPTION
**Pull request by jtvmatos**
This PR solve the issue describe in https://github.com/OpenAPITools/openapi-generator/issues/13129.
The deepcopy function throws a ApiValueError (`Cannot deserialize input data due to missing discriminator. The discriminator property 'className' is missing at path: ()`) for some models with AllOf in schema e.g.,  `from petstore_api.model.dog import Dog`.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.1.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Python technical committee:
@taxpon @frol @mbohlool @cbornet  @kenjones-cisco @tomplus @arun-nalla @spacether

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/python/model_templates/methods_shared.mustache [hunk 1]
```diff
@@ -27,7 +27,7 @@
         if self.get("_spec_property_naming", False):
             new_inst = cls._new_from_openapi_data()
         else:
-            new_inst = cls.__new__(cls)
+            new_inst = cls.__new__(cls, **self.__dict__)
```
Comment (spacether) [inline modules/openapi-generator/src/main/resources/python/model_templates/methods_shared.mustache:30]: But line 33 below assigns properties
Why is this needed?

Comment (jtvmatos) [reply modules/openapi-generator/src/main/resources/python/model_templates/methods_shared.mustache:30]: Hi spacether, 

I don't know if this solution is the best. But without this change the example/test doesn't work.
```
>>> from copy import deepcopy
>>> from petstore_api.model.dog import Dog
>>> model_dog = Dog(class_name='Dog', color='white', breed='Jack Russel Terrier')
>>> new_dog = deepcopy(model_dog)

petstore_api.exceptions.ApiValueError: Cannot deserialize input data due to missing discriminator. The discriminator property 'className' is missing at path: ()
```
The error occurs inside the deepcopy loop, when running the `new_inst = cls.__new__(cls)` operation for one of the composed schemas (`<class 'petstore_api.model.animal.Animal'>` class). In this step of the loop, the `discriminator` value is `{'class_name': {'Cat': <class 'petstore_api.model.cat.Cat'>, 'Dog': <class 'petstore_api.model.dog.Dog'>}}`, and without kargs to escape in the `cls` in `kwargs.get('_visited_composed_classes', ())` (L-231-235 model_utils.py) the deepcopy fail in L-266 (model_utils.py).

Comment (jtvmatos) [reply modules/openapi-generator/src/main/resources/python/model_templates/methods_shared.mustache:30]: hi @spacether! 

Do I need to provide more info?

Thanks!

Comment (spacether) [reply modules/openapi-generator/src/main/resources/python/model_templates/methods_shared.mustache:30]: Nope no need, th

**Review by spacether — APPROVED** [review]
Looks good, thank you for the fix!
